moe.karaokes.mugen.app.yml

```diff
@@ -0,0 +1,93 @@
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+command: km-wrapper
+separate-locales: false
+finish-args:
+  - '--filesystem=host'
+  - '--share=ipc'
+  - '--socket=x11'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--device=dri'
+  - '--share=network'
+  - '--system-talk-name=org.freedesktop.UPower'
+  - '--talk-name=com.canonical.AppMenu.Registrar'
+  - '--talk-name=com.canonical.dbusmenu'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+modules:
+  - name: karaokemugen
+    buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/node16/bin
+      env:
+        XDG_CACHE_HOME: /run/build/karaokemugen/flatpak-node/cache
+        npm_config_nodedir: /usr/lib/sdk/node16
+    build-commands:
+      - install -D sentry-cli-Linux-x86_64 /app/bin/sentry-cli
+      - >-
+        HOME=$PWD yarn config --offline set yarn-offline-mirror
+        $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror
+      - SENTRYCLI_USE_LOCAL=1 yarn --offline
+      - yarn --offline build
+      - yarn --offline installkmfrontend
+      - yarn --offline buildkmfrontend
+      - git rev-parse HEAD | head -c 8 > assets/sha.txt
+      - >-
+        . flatpak-node/electron-builder-arch-args.sh; yarn run electron-builder
+        --dir -c.compression=store -c.mac.identity=null
+        $ELECTRON_BUILDER_ARCH_ARGS
+      - cp -r packages/linux*unpacked /app/km
+      - install -d /app/km/app/
+      - cp -r bin/ /app/km/app/
+      - install -Dm755 stub_sandbox /app/km/chrome-sandbox
+      - install -Dm755 km -t /app/bin/
+      - install -Dm755 km-wrapper /app/bin/km-wrapper
+      - install -d /app/share/applications
+      - >-
+        install -Dm644 moe.karaokes.mugen.app.desktop
+        "/app/share/applications/${FLATPAK_ID}.desktop"
+      - >-
+        install -Dm644 moe.karaokes.mugen.app.png
+        "/app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png"
+      - >-
+        install -Dm644 moe.karaokes.mugen.app.metainfo.xml
+        /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: git
+        url: https://gitlab.com/karaokemugen/karaokemugen-app
+        commit: e4452fcaa5a017205390225ba832aeb3a0944fa3
+        tag: v7.0.37
+      - generated-sources.json
+      - type: script
+        dest-filename: km
+        commands:
+          - export TMPDIR=$XDG_RUNTIME_DIR/app/$FLATPAK_ID
+          - zypak-wrapper /app/km/karaokemugen "$@"
+      - type: script
+        dest-filename: stub_sandbox
+        commands:
+          - |
+            echo Stub sandbox ignoring command: $@
+            exit 1
+      - type: file
+        path: km-wrapper
+      - type: file
+        path: moe.karaokes.mugen.app.metainfo.xml
+      - type: file
+        path: moe.karaokes.mugen.app.png
+      - type: file
+        path: moe.karaokes.mugen.app.desktop
```

Comment:
doesn't electron builder generate one?

Reply:
Electron-builder does generate one when you specify a `deb` or other packed targets. When building into an unpacked folder it doesn't generate a `.desktop` file. 

I've seen other apps like codium build a `.deb` with electron-builder and extract its contents to get the `.desktop` file. Should we do that as well instead of providing it here? See [here](https://github.com/flathub/com.vscodium.codium/blob/c9cfa14dd1c71d5dce6e42bf47e975dbce84f2f0/com.vscodium.codium.yaml#L70)

Some other repos like [Visual Studio Code](https://github.com/flathub/com.visualstudio.code/blob/master/com.visualstudio.code.desktop) provide their own, so we thought it was okay.

Reply:
I removed the png from the repo but left the .desktop file, is that okay?